### ISSUE TITLE
Overlay not working

### ISSUE THREAD

**Issue by Sabin-Adhikari** [maitrungduc1410/primeng-shadowdom-directives#14]
Hi , Thanks for your amazing library.
I am having issues working with PConfirmPopup and other overlay controls.
It seems like the confirmationService or overlay service is injecting the control to main document (which does not contain any stylesheet) which is why the overlay does not have any styles. I have tried using the directive you provided in the library but it won't work. In your demo app, you are linking the style.css to the index.html which is why the overlay is working . 
I need to encapsulate the entire CSS of my web component so linking is not a solution for me. 
Please help!
<img width="424" alt="image" src="https://github.com/maitrungduc1410/primeng-shadowdom-directives/assets/87262876/03180703-a869-4f6a-aad3-660d80ed7471">

Here the div container for overlay or popConfirm is appended to the main body instead of shadowRoot.
<img width="948" alt="image" src="https://github.com/maitrungduc1410/primeng-shadowdom-directives/assets/87262876/01e5d61a-5b9d-40ad-a953-c7ebb715d624">

I am just sending a click event to this function and rendering the pop confirm: 

```js
   confirmDelete(event: Event,node) {
      event.preventDefault()
      event.stopPropagation()
      this.confirmationService.confirm({
          target: event.target as EventTarget,
          message: 'Are you sure that you want to delete this page?',
          icon: 'pi pi-exclamation-triangle',
          accept: () => {
     
          }
    });

```

**Comment by Sabin-Adhikari** [comment]
Also, I am using following versions in my app.
```json
 "@angular/animations": "^16.2.0",
    "@angular/common": "^16.2.0",
    "@angular/compiler": "^16.2.0",
    "@angular/core": "^16.2.0",
    "@angular/elements": "^16.2.5",
    "@angular/forms": "^16.2.0",
    "@angular/platform-browser": "^16.2.0",
    "@angular/platform-browser-dynamic": "^16.2.0",
    "@angular/router": "^16.2.0",
    "primeicons": "^6.0.1",
    "primeng": "^16.3.1",
    "primeng-shadowdom-directives": "^0.2.2",
    "rxjs": "~7.8.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.13.0"`
```

**Comment by maitrungduc1410** [comment]
This is issue not related to the problem my lib solves.

Your issue is because you're trying to append the overlay component to body. Meanwhile PrimeCSS is not available at document root context.

Thats why in my demo you see I need to have PrimeCss in styles.scss to make sure PrimeNG style is global for this case.

There's no choice :)

**Comment by Sabin-Adhikari** [comment]
Is there any way to resolve this? 
Any place that you can point me at to fix this?

I thought if I use the pConfirmPopup inside my app.component (which has ShadowDOM view encapsulation) ..it would append it to the shadow root ..But it adds it to the main body

Any help is appreciated.

**Comment by maitrungduc1410** [comment]
You have the choice of not using "appendTo=body" then it'll be rendered inside shadowDom. In that case you can do import style directly from your component file with "styleUrls"

**Comment by Sabin-Adhikari** [comment]
How do I append it to shadow DOM?  The confirmation service is always adding it to the main body

**Comment by maitrungduc1410** [comment]
Yes, all overlay components have "appendTo" pls check PrimeNG docs

https://primeng.org/dropdown#api.dropdown.props.appendTo

**Comment by maitrungduc1410** [comment]
From what I know, default it should render inside shadowDOM

You can run example project in this repo and check

**Comment by Sabin-Adhikari** [comment]
It seems like they don't have the appendTo for pConfirmDialog..I'm not sure if there's any fix for this

**Comment by maitrungduc1410** [comment]
Confirm dialog does have bro

https://primeng.org/confirmdialog#api.confirmdialog.props.appendTo

**Comment by Sabin-Adhikari** [comment]
I am using confirmPopup. Sorry for the confusion.
https://primeng.org/confirmpopup

I can't find its appendTo

**Comment by Sabin-Adhikari** [comment]
I tried rewriting its implementation but tough luck..I can see the control puts it to the document body in confirmPopup.ts file.
```js
    onAnimationStart(event: AnimationEvent) {
        if (event.toState === 'open') {
            this.container = event.element;
            this.renderer.appendChild(this.document.body, this.container);
            this.align();
            this.bindListeners();

            const element = this.getElementToFocus();
            if (element) {
                element.focus();
            }
        }
    }
` ``

**Comment by Sabin-Adhikari** [comment]
Would you be able to help fix this?

**Comment by maitrungduc1410** [comment]
Hi @Sabin-Adhikari you better raise a PR in PrimeNG and ask for this. Or implement your own, but not recommended.

**Comment by Sabin-Adhikari** [comment]
Thanks a lot man! I got it working finally after using appendTo and custom directives .

**Comment by Sabin-Adhikari** [comment]
Resolved! 
I made a custom directive for tooltips (from scratch) and pop confirms (using prime ng overlay and appendTo).
Then I defined a div element template references in the root application, and passed the reference to downstream components. Any components using the Overlays pass this reference in appendTo. 

This ensures any overlays are appended correctly inside the shadowDOM.

PS: my root application has shadowDOM while child components have Emulated encapsulation.
